DEVELOPMENT.md

```diff
@@ -0,0 +1,219 @@
+
+
+## Run & debug
+
+To run an instance of IDE with Git Machete IntelliJ Plugin installed from the current source,
+execute `:runIde` Gradle task (`Gradle panel` -> `Tasks` -> `intellij` -> `runIde` or `./gradlew runIde`).
+
+To watch the logs of this IntelliJ instance, run `tail -f build/idea-sandbox/system/log/idea.log`.
+
+
+## Run UI tests
+
+```
+./scripts/run-ui-tests [<intellij-version>]
+```
+
+See [Gradle Intellij plugin docs](https://github.com/JetBrains/gradle-intellij-plugin/tree/master/examples/ui-test-example)
+for more details.
+
+
+## Generate plugin zip
+
+To generate a plugin archive run `:buildPlugin` Gradle task (`Gradle panel` -> `Tasks` -> `intellij` -> `buildPlugin` or `./gradlew buildPlugin`).<br/>
+The resulting file will be available under `build/distributions/`.
+
+
+## Install snapshot build of the plugin from CI
+
+Download the zip file from the artifacts of the given build in [CircleCI](https://app.circleci.com/pipelines/github/VirtusLab/git-machete-intellij-plugin). <br/>
+Go to `File` -> `Settings` -> `Plugins` -> `(gear icon)` -> `Install Plugin from Disk...`, select the zip and restart IDE.
+
+
+## Logging
+
+SLF4J logging in this plugin has the following categories:
+
+```
+binding
+branchlayout
+gitcore
+gitmachete.backend
+gitmachete.frontend.actions
+gitmachete.frontend.externalsystem
+gitmachete.frontend.graph
+gitmachete.frontend.ui
+```
+
+The standard practice in Java logging is to use fully-qualified name of each class as category;
+in our case, however, we're only ever using the categories provided above.
+FQCN (and method name), however, is always a part of the log message itself.
+
+By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
+The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
+Tip: use `tail -f` to watch the log file as it grows.
+
+To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
+A relatively small amount of `TRACE`-level logs is generated as well.
+
+
+## Coding conventions
+
+Most non-standard/project-specific conventions are enforced by:
+
+* [pre-commit hook](scripts/run-pre-build-checks)
+* [Spotless](https://github.com/diffplug/spotless/tree/master/plugin-gradle) for Java code formatting (see [Eclipse-compatible config](config/spotless/formatting-rules.xml))
+* [Checkstyle](https://checkstyle.sourceforge.io/) for code style/detecting basic smells (see [top-level config](config/checkstyle/checkstyle.xml))
+* [Checker Framework](https://checkerframework.org/manual/) for formal correctness, esp. wrt. null safety and UI thread handling
+  (most config in [build.gradle](build.gradle), stubs in [config/checker/](config/checker))
+
+Note that certain Checker Framework's checkers (`Index`, `Optional`, `Regex`) are by default only run in the CI to speed up local compilation. <br/>
```

Comment:
Whoops this is also outdated, just remove it completely (already documented under `Build`)